### PR TITLE
python3Packages.qiskit: 0.17.0 -> 0.18.2

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change
Update qiskit to latest version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

``python38Packages.qiskit`` is not available due to upstream breakages in ``master`` (I think ``pythonPackages.aiohttp -> ... -> pythonPackages.qiskit-ibmq-provider``). Building ignis builds [terra, air, ignis], so it's a good enough proxy.

**Comment by drewrisinger** [comment]
``nixpkgs-review pr 84943``:
https://github.com/NixOS/nixpkgs/pull/84943
26 package marked as broken and skipped:
gdata-sharp iterm2 linuxPackages_4_4.evdi linuxPackages_hardkernel_4_14.bcc linuxPackages_hardkernel_4_14.bpftrace linuxPackages_hardkernel_4_14.can-isotp linuxPackages_hardkernel_4_14.chipsec linuxPackages_hardkernel_4_14.digimend linuxPackages_hardkernel_4_14.evdi linuxPackages_hardkernel_4_14.mba6x_bl linuxPackages_hardkernel_4_14.nvidia_x11 linuxPackages_hardkernel_4_14.nvidia_x11_legacy390 linuxPackages_hardkernel_4_14.nvidiabl linuxPackages_hardkernel_4_14.prl-tools newlibCross octave-jit php73Extensions.zmq php74Extensions.couchbase python27Packages.caffe python27Packages.habanero python27Packages.handout python27Packages.hass-nabucasa python37Packages.pyblock python38Packages.nixpart python38Packages.poezio python38Packages.qiskit-ibmq-provider

16 package built:
python27Packages.python-constraint python37Packages.fastjsonschema python37Packages.python-constraint python37Packages.qasm2image python37Packages.qiskit python37Packages.qiskit-aer python37Packages.qiskit-aqua python37Packages.qiskit-ibmq-provider python37Packages.qiskit-ignis python37Packages.qiskit-terra python38Packages.fastjsonschema python38Packages.python-constraint python38Packages.qiskit-aer python38Packages.qiskit-aqua python38Packages.qiskit-ignis python38Packages.qiskit-terra

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

**Comment by drewrisinger** [comment]
Last push was just a cleanup/meta change, all packages have same hash after.

**Comment by drewrisinger** [comment]
Update: added retworkx from #84945 (figured out how to get it stable), and then force-pushed.
@GrahamcOfBorg build python37Packages.retworkx python38Packages.retworkx
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

**Comment by FRidh** [comment]
Some build is still failing.

**Comment by drewrisinger** [comment]
@FRidh ``scipy`` build is failing (#85637) due to #83888 (#85636 will resolve).
When branch is checked out locally (rebased on commit prior to #83888 merge), it builds fine.

Relevant section from borg log:
```
numpy.distutils.system_info.NotFoundError: No lapack/blas resources found.
builder for '/nix/store/k55al0ykln7iszalhv6qm8fsjjasvgif-python3.8-scipy-1.4.1.drv' failed with exit code 1
```

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

Result of running ``nixpkgs-review pr 84943``:
https://github.com/NixOS/nixpkgs/pull/84943
18 package built:
python27Packages.python-constraint python37Packages.fastjsonschema python37Packages.python-constraint python37Packages.qasm2image python37Packages.qiskit python37Packages.qiskit-aer python37Packages.qiskit-aqua python37Packages.qiskit-ibmq-provider python37Packages.qiskit-ignis python37Packages.qiskit-terra python37Packages.retworkx python38Packages.fastjsonschema python38Packages.python-constraint python38Packages.qiskit-aer python38Packages.qiskit-aqua python38Packages.qiskit-ignis python38Packages.qiskit-terra python38Packages.retworkx

**Comment by drewrisinger** [comment]
fixed a minor reference issue (from comments only) & updated to latest master version.
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

**Comment by drewrisinger** [comment]
Added update of qiskit-ibmq-provider
@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit-ignis

File: pkgs/development/python-modules/retworkx/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    rev = rx-version;
+    sha256 = "160w5vkzrl5rzcrdwhjq820i5lmc527m6hg0kxx0k6n2bz9qn26g";
+  };
+in
+buildPythonPackage rec {
+  pname = "retworkx";
+  version = rx-version;
+  format = "wheel";
+
+  disabled = pythonOlder "3.5" || pythonAtLeast "3.9"; # compiled versions only included for 3.5 <= py <= 3.8
```
Comment (bhipple) [inline pkgs/development/python-modules/retworkx/default.nix:33]: Ideally we would just cleanly compile this thing entirely from source:
https://github.com/Qiskit/retworkx#building-from-source

but it looks like that might be a project in and of itself with the Rust bindings.

**Review by bhipple — APPROVED** [review]

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 84943` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>18 packages built:</summary>
<br>- python27Packages.python-constraint
<br>- python37Packages.fastjsonschema
<br>- python37Packages.python-constraint
<br>- python37Packages.qasm2image
<br>- python37Packages.qiskit
<br>- python37Packages.qiskit-aer
<br>- python37Packages.qiskit-aqua
<br>- python37Packages.qiskit-ibmq-provider
<br>- python37Packages.qiskit-ignis
<br>- python37Packages.qiskit-terra
<br>- python37Packages.retworkx
<br>- python38Packages.fastjsonschema
<br>- python38Packages.python-constraint
<br>- python38Packages.qiskit-aer
<br>- python38Packages.qiskit-aqua
<br>- python38Packages.qiskit-ignis
<br>- python38Packages.qiskit-terra
<br>- python38Packages.retworkx
</details>

File: pkgs/development/python-modules/retworkx/default.nix (hunk 1 above)
Comment (drewrisinger) [reply pkgs/development/python-modules/retworkx/default.nix:33]: Yeah, I tried doing that in https://github.com/NixOS/nixpkgs/pull/84945 before I gave up. It gets messy and requires a lot of other modules, which themselves require the Rust nightly build which then requires an overlay of nixpkgs...

**Comment by drewrisinger** [comment]
Thanks for the feedback @bhipple!

**Pull request merged**